crates/sui-storage/src/event_store/sql.rs

```diff
@@ -168,131 +187,192 @@ impl SqlEventStore {
-    if let Some(json_value) = &event.move_struct_json_value {
-        json_value.to_string()
-    } else {
-        let maybe_json = match &event.event {
-            Event::TransferObject {
-                version,
-                recipient,
-                type_,
-                ..
-            } => Some(json!({"destination": recipient.to_string(),
-                       "version": version.value(),
-                       "type": type_.to_string() })),
-            // TODO: for other event types eg EpochChange
-            _ => None,
+impl From<SqliteRow> for StoredEvent {
+    // Translate a Row into StoredEvent
+    // TODO: convert to use FromRow trait so query_as() could be used?
+    // TODO: gracefully handle data corruption/incompatibility without panicking
+    fn from(row: SqliteRow) -> Self {
+        let timestamp: i64 = row.get(EventsTableColumns::Timestamp as usize);
+        let checkpoint: i64 = row.get(EventsTableColumns::Checkpoint as usize);
+        let digest_raw: Option<Vec<u8>> = row.get(EventsTableColumns::TxDigest as usize);
+        let tx_digest = digest_raw.map(|bytes| {
+            TransactionDigest::new(
+                bytes
+                    .try_into()
+                    .expect("Error converting digest bytes to TxDigest"),
+            )
+        });
+        let event_type: u16 = row.get(EventsTableColumns::EventType as usize);
+        let package_id =
+            SqlEventStore::try_extract_object_id(&row, EventsTableColumns::PackageId as usize)
+                .expect("Error converting stored package ID bytes to ObjectID");
+        let object_id =
+            SqlEventStore::try_extract_object_id(&row, EventsTableColumns::ObjectId as usize)
+                .expect("Error converting stored object ID bytes to ObjectID");
+        let module_name: Option<String> = row.get(EventsTableColumns::ModuleName as usize);
+        let function: Option<String> = row.get(EventsTableColumns::Function as usize);
+        let fields_text: &str = row.get(EventsTableColumns::Fields as usize);
+        let fields: Vec<_> = if fields_text.is_empty() {
+            Vec::new()
+        } else {
+            let fields_json = serde_json::from_str(fields_text)
+                .unwrap_or_else(|e| panic!("Could not parse [{}] as JSON: {}", fields_text, e));
+            if let Value::Object(map) = fields_json {
+                map.into_iter()
+                    .map(|(k, v)| (flexstr::SharedStr::from(k), EventValue::Json(v)))
+                    .collect()
+            } else {
+                warn!(
+                    ?fields_json,
+                    "Could not parse JSON as object, should not happen"
+                );
+                Vec::new()
+            }
         };
-        maybe_json.map(|j| j.to_string()).unwrap_or_default()
+        let move_event_contents: Option<Vec<u8>> = row.get(EventsTableColumns::Contents as usize);
+        let move_event_struct_tag: Option<String> =
```

Comment:
Can we call this `move_event_type_name` or `move_event_name`?

Reply:
plz see https://github.com/MystenLabs/sui/pull/3989#discussion_r947429911